ee/tabby-schema/src/schema/web_documents.rs

```diff
@@ -0,0 +1,81 @@
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use juniper::{GraphQLInputObject, GraphQLObject, ID};
+use validator::Validate;
+
+use crate::{job::JobInfo, juniper::relay::NodeType, Context, Result};
+
+#[derive(GraphQLObject)]
+#[graphql(context = Context)]
+pub struct CustomWebDocument {
+    pub url: String,
+    pub name: String,
+    pub id: ID,
+    pub created_at: DateTime<Utc>,
+    pub job_info: JobInfo,
+}
+
+#[derive(GraphQLObject)]
+#[graphql(context = Context)]
+pub struct PresetWebDocument {
+    pub name: String,
+    pub id: ID,
+    pub job_info: JobInfo,
+}
+
+impl CustomWebDocument {
+    pub fn source_id(&self) -> String {
+        Self::format_source_id(&self.id)
+    }
+
+    pub fn format_source_id(id: &ID) -> String {
+        format!("web_document:{}", id)
+    }
+}
+
+#[derive(Validate, GraphQLInputObject)]
+pub struct CreateCustomDocumentInput {
+    #[validate(url(code = "name", message = "Invalid Name"))]
+    pub name: String,
+    #[validate(url(code = "url", message = "Invalid URL"))]
+    pub url: String,
+}
+
+#[derive(Validate, GraphQLInputObject)]
+pub struct SetPresetDocumentActiveInput {
+    #[validate(url(code = "name", message = "Invalid Name"))]
+    pub name: String,
```

Comment:
```suggestion
    pub id: ID,
```

since we have `id` in `preset_web_documents` interface, let's just use it here.

Reply:
No. Only active `PresetWebDocument` will be assigned an ID.

Reply:
It is converted from result of `INSERT` sql

Reply:
ok - we can leave the discussion to the implementation PR